### PR TITLE
Pzprxs merge

### PR DESCRIPTION
**Pull request by robx**
This is a merge of current pzprxs master branch. The main changes vs. pzprxs are removing the vercel-related things, and a bit of build stuff (Gruntfile etc.). For the moment mostly trying to see if things just work, or if there's more to do.

### CODE REVIEW

**Comment by robx** [comment]
(It might not be necessary to drop the vercel support things, just didn't want that to interfere.)

**Comment by robx** [comment]
This was rather smooth -- merged version is currently served temporarily at https://puzz.link/pzprxs/ (e.g. https://puzz.link/pzprxs/list.html, https://puzz.link/pzprxs/p.html; the links to `/p?type` won't work).

Some things to deal with / think about before maybe putting this live as the main player on puzz.link. On an explicit, technical level:
- list.html is the pzprxs-only list right now
- "About" references pzprxs, needs some kind of update

And the less technical points:
- I'm a bit unsure about the impact of publishing a lot of new, niche puzzle types
  - would be nice to have some kind of tier system (as with original pzprjs / nikoli) and have types prove themselves maybe hide some of the weird old types too in some fashion?)
  - it might be an option to promote types here only after they've proved themselves for a while elsewhere
  - to some extent this isn't so much a player applet question, and rather has to do with how the types show up in the database -- the puzzle type list would balloon, might be necessary to filter that list...
- I'm a bit uncomfortable with the puzzle type credit -- I'm all for crediting inventors, but don't want to be implicated in judging origin; e.g. the current list credits some sole inventors of types that would really deserve to credit some of the puzzle types they're based on. I'd rather not be involved in topics like that, would be a lot more comfortable linking to some external resource(s) that can expound on the history

File: src/pzpr/variety.js [hunk 1]
```diff
@@ -45,8 +46,75 @@
...
+	 */
+	var inhibited = false;
+	if (pzpr.env.browser) {
+		inhibited =
+			window.location.protocol !== "file:" &&
+			window.location.host.indexOf("localhost") === -1;
+	} else if (pzpr.env.node) {
+		inhibited = process.env.VERCEL;
+	}
+	var allowedGenres = new Set([
```
Comment (x-sheep) [inline src/pzpr/variety.js:65]: Removing this section and all references will enable every genre in the list.

**Comment by x-sheep** [comment]
I would be very happy to roll everything into puzz.link. I've removing some references to puzz.link in my fork such as Network Play, which need to be put back into place. There's also some bonus stuff which need your help to be activated here:

* Previews for editor urls such as https://pzprxs.vercel.app/pv?anglers
* Registering with Codecov (very helpful for new PR's even if the project coverage is not 100%)
* Installing Weblate on this repo as a GitHub app, so I can point the existing project to here without starting over.

I've destroyed the category system in the list.html, which was already woefully out of date on puzz.link itself i.e. Nikoli types having been promoted or dropped, and an increasing number of genres not related to Nikoli which fall outside all existing categories. It would be helpful to get more community input on how to categorize everything.

Lastly, the primary thing that will lead me to sunset pzprxs and redirect everything to puzz.link is the ability to deploy to puzz.link by myself. More specifically, being able to hand deployment rights to other people just by giving them Maintainer role on GitHub, in case either of us gets too busy to keep up with changes. This is what Vercel gave me out of the box, and I'm looking forward to your implementation in #265.

File: src/pzpr/variety.js [hunk 2]
```diff
@@ -46,76 +46,8 @@
...
-		"koburin",
-		"mirrorbk"
-	]);
-
 	(function(Genre, obj) {
 		for (var pzprid in obj) {
-			if (inhibited && !allowedGenres.has(pzprid)) {
-				continue;
-			}
-			_info[pzprid] = new Genre(pzprid, obj[pzprid]);
```
Comment (x-sheep) [inline src/pzpr/variety.js:50]: This line should still be there.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -1,5 +1,8 @@
 on:
-  pull_request
+  pull_request:
+  push:
+    branches:
+      - master
```
Comment (x-sheep) [inline .github/workflows/test.yml:5]: This should be `main`.

**Comment by x-sheep** [comment]
There's one more thing to consider with the image preview. The Piece Bank can be conditionally shown or hidden. For certain configurations, the bank is hidden by default. In this case, a Pentopia puzzle with the default Pentomino set does not require a preview:

https://pzprxs.vercel.app/p?pentopia/10/10/zzjfzzu//p

Other configurations do default to having a preview:

https://pzprxs.vercel.app/p?pentopia/10/10/zzjfzzu//d

The exact change needed in `pzprnode` is to override the bank display. `bank` forces it to be shown, while `no-bank` forces it to be hidden. e.g. `https://pzprxs.vercel.app/pv?no-bank&pentopia/10/10/zzjfzzu//d`

**Comment by robx** [comment]
> The exact change needed in `pzprnode` is to override the bank display. `bank` forces it to be shown, while `no-bank` forces it to be hidden. e.g. `https://pzprxs.vercel.app/pv?no-bank&pentopia/10/10/zzjfzzu//d`

I've implemented this (with a slight change to the arguments) over here: https://github.com/robx/pzprnode/pull/15, could you check if that makes sense?

**Pull request merged**